code/src/components/TodoItem.js

```diff
@@ -0,0 +1,62 @@
+import React from 'react'
+import styled from 'styled-components'
+import { useSelector, useDispatch } from 'react-redux'
+import { todos } from '../reducers/todos'
+import { CustomCheckbox } from './CustomCheckbox'
+
+const TodoListItem = styled.li`
+list-style: none;
+font-size: 20px;
+display: flex;
+align-items: center;
+margin-bottom: 10px;
+`
+const ItemText = styled.span`
+margin-left: 10px;
+font-size: 18px;
+color: #24292e;
+`
+const RemoveLink = styled.a`
+font-size: 12px;
+margin-left: 5px;
+color: #24292e;
+cursor: pointer;
+
+&:hover {
+  color: blue;
+}
+`
+
+export const TodoItem = ({ itemIndex }) => {
+  // Get the item from the store based on the index
+  const item = useSelector(store => store.todos.list.items[itemIndex])
+
+  const dispatch = useDispatch()
+
+  // Function to handle remove item
+  const onRemoveClicked = event => {
+    dispatch(
+      todos.actions.removeTodo({
+        itemIndex: itemIndex
+      })
+    )
+  }
```

Comment:
In the case you were not using Redux here, you would have to pass a function into this component which allows you to modify some state in the parent component.

Since your parent component is App.js, the list of Todo items would be a state in that component.  

Essentially App.js would become quite complex very quickly because you would have to define a function for any action you wanted to do in the list, and send that function to a child component.  

If you have about 30 functions, things will get very confusing.  Having all of the actions specified in the same place would make that scenario more understandable and manageable.